TPL/force-app/main/default/classes/HealthcareCostTriggerHandler.cls

```diff
@@ -1,165 +1,50 @@
-                }
-                if(recordType == 'Future_Care'){
-                    c.Future_Care_Total2__c = tempMap.get(recordType);
-                }
-            }    
-            Case c1 = caseUpdateMap.get(c.Id);
-            c1.Continuing_Care_Total2__c = c.Continuing_Care_Total2__c;
-            c1.Future_Care_Total2__c = c.Future_Care_Total2__c;
-            caseUpdateMap.put(c1.id,c1);
-        }
-        try{
-            update caseUpdateMap.values();
-        }catch(Exception e){
-            System.debug(e.getMessage());
-        }
-        
-    } 
-    public static void updateRollupDelete(List<Healthcare_Cost__c> hccList){
-        System.debug(hccList);
-        set<Id> caseIds = new set<Id>();
-        for(Healthcare_Cost__c hcc : hccList){
-            caseIds.add(hcc.Case2__c);
-        }
-        System.debug(caseIds);
-        List<Case> caseList = [select id,Ambulance_Total2__c,Pharmacare_Total2__c,Hospitalization_Total2__c,MSP_Total2__c,Continuing_Care_Total2__c,Future_Care_Total2__c from case where id IN: caseIds];
-        Map<Id,Case> caseMap = new Map<Id,Case>();
-        for(Case c : caseList){
-            caseMap.put(c.id,c);
-        }
-        for(Healthcare_Cost__c hcc : hccList){
-            if(hcc.RecordTypeName__c != 'Continuing_Care' && hcc.RecordTypeName__c != 'Future_Care'){
-                
-                if(hcc.Cost_Include__c && hcc.Cost_Review__c){
-                    Case c = caseMap.get(hcc.Case2__c);
-                    if(hcc.RecordTypeName__c == 'Ambulance'){
-                        c.Ambulance_Total2__c = c.Ambulance_Total2__c - hcc.Total_Cost_Override__c;
-                    }
-                    if(hcc.RecordTypeName__c == 'Pharmacare'){
-                        c.Pharmacare_Total2__c = c.Pharmacare_Total2__c - hcc.Total_Cost_Override__c;
-                    }
-                    if(hcc.RecordTypeName__c == 'Hospitalization'){
-                        c.Hospitalization_Total2__c = c.Hospitalization_Total2__c - hcc.Total_Cost_Override__c;
-                    }
-                    if(hcc.RecordTypeName__c == 'MSP'){
-                        c.MSP_Total2__c = c.MSP_Total2__c - hcc.Total_Cost_Override__c;
-                    }
-                    caseMap.put(c.Id,c);
-                    
-                }
+            if(caseMap.containsKey((String)agg.get('case2__c'))){
+                c = caseMap.get((String)agg.get('case2__c'));
             }else{
-                System.debug('149');
-                Case c = caseMap.get(hcc.Case2__c);
-                if(hcc.RecordTypeName__c == 'Continuing_Care'){
-                    c.Continuing_Care_Total2__c = c.Continuing_Care_Total2__c - hcc.Cost__c;
-                }
-                if(hcc.RecordTypeName__c == 'Future_Care'){
-                    c.Future_Care_Total2__c = c.Future_Care_Total2__c - hcc.Cost__c;
-                }
-                caseMap.put(c.Id,c);	
+                c.id = (String)agg.get('case2__c');
```

Comment:
Note to team: I approved this rollback to the original process after Ahmed worked on the solution using the original planned mechanism (aggregate queries) -- this is the core change where the default original values are set prior to processing the query results.